src/ToolBox/SOS/NETCore/SymbolReader.cs

```diff
@@ -389,23 +389,30 @@ internal static bool GetLocalVariableByIndex(IntPtr symbolReaderHandle, int meth
         /// <remarks>used by the gdb JIT support (not SOS). Does not support in-memory PEs or PDBs</remarks>
         internal static bool GetInfoForMethod(string assemblyPath, int methodToken, ref MethodDebugInfo debugInfo)
         {
-            List<DebugInfo> points = null;
-
-            if (!GetDebugInfoForMethod(assemblyPath, methodToken, out points))
+            try
             {
-                return false;
-            }
-            var structSize = Marshal.SizeOf<DebugInfo>();
+                List<DebugInfo> points = null;
+
+                if (!GetDebugInfoForMethod(assemblyPath, methodToken, out points))
+                {
+                    return false;
+                }
+                var structSize = Marshal.SizeOf<DebugInfo>();
 
-            debugInfo.size = points.Count;
-            var ptr = debugInfo.points;
+                debugInfo.size = points.Count;
+                var ptr = debugInfo.points;
 
-            foreach (var info in points)
+                foreach (var info in points)
+                {
+                    Marshal.StructureToPtr(info, ptr, false);
+                    ptr = (IntPtr)(ptr.ToInt64() + structSize);
+                }
+                return true;
+            }
+            catch
             {
-                Marshal.StructureToPtr(info, ptr, false);
-                ptr = (IntPtr)(ptr.ToInt64() + structSize);
             }
-            return true;
+            return false;
         }
```

Comment:
Does this part of the change relate to 16 byte data type issue? I'm not opposed to it but I couldn't figure out how it related.